### PR TITLE
MOBILE-159: Feed section phase 1.2 Wrap up

### PR DESCRIPTION
**Pull request by 07jasjeet**
Not being tracked by an exact ticket but is an extension of MOBILE-159 and covers one small task:
- [X] Updated View-model tests.

### CODE REVIEW

File: app/src/main/java/org/listenbrainz/android/di/ServiceModule.kt [hunk 1]
```diff
@@ -35,7 +35,6 @@ class ServiceModule {
     
     
     @get:Provides
-    @get:Singleton
```
Comment (akshaaatt) [inline app/src/main/java/org/listenbrainz/android/di/ServiceModule.kt:38]: Why would you not keep the services Singleton anymore?

File: app/src/main/java/org/listenbrainz/android/viewmodel/SearchViewModel.kt [hunk 2]
```diff
@@ -184,4 +190,9 @@ class SearchViewModel @Inject constructor(
             inputQueryFlow.emit("")
         }
     }
+    
+    /** True if the error states the the user is already being followed.*/
+    private fun userIsAlreadyFollowed(error: ResponseError?): Boolean =
+        error == ResponseError.BAD_REQUEST &&
+                error.actualResponse?.contains("already following") == true
```
Comment (akshaaatt) [inline app/src/main/java/org/listenbrainz/android/viewmodel/SearchViewModel.kt:197]: avoid this as said before

**Review by akshaaatt — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/listenbrainz/android/viewmodel/SearchViewModel.kt (hunk 2 above)
Comment (07jasjeet) [reply app/src/main/java/org/listenbrainz/android/viewmodel/SearchViewModel.kt:197]: This is a temporary solution until we the api itself is updated or we have profile screen in the app so we can remove the follow button all together.

File: app/src/main/java/org/listenbrainz/android/di/ServiceModule.kt (hunk 1 above)
Comment (07jasjeet) [reply app/src/main/java/org/listenbrainz/android/di/ServiceModule.kt:38]: Will ping you on IRC for this one.

**Pull request merged**